### ISSUE TITLE
[expo-yarn-workspaces] Add web support

### ISSUE THREAD

**Issue by EvanBacon** [expo/expo#7442]
In this repo we currently use a few different things to make our yarn workspaces work properly. It'd be better to create a web version of `expo-yarn-workspaces` that we could export from a project's `webpack.config.js`.

**Comment by itsJASPERr** [comment]
Quick question, because I have not duck very deep yet, neither have I found a related Issue.
Generally, yarn workspaces work for web for me so far, but it seems like it has an issue with Typescript. No problems compiling for device. Do you have any thoughts on this? Anything is appreciated, I can understand if this way to little info.

```
Module parse failed: Unexpected token (4:7)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
| import playground from "./playground";
| 
> export interface PaletteIntention {
|   light?: string;
|   main: string;
```

**Comment by andrioid** [comment]
I have a React-Native-Web app that I wanted to move to Expo, but came across an interface error like @itsJASPERr 

Can you elaborate on what we need to make this work for web (either in this module, or as a seperate thing)? I'd be willing to put a few hours in.

**Comment by itsJASPERr** [comment]
I think that issue is solved for me somehow...I think it had to do with how yarn workspaces solves local dependencies. I believe what should help is 
- make sure to have the dependencies of the other workspace packages in your package.json
- either try `yarn install --focus` to package dependencies in that workspace
- publish the packages to a private package registry and let expo pull it from there

I am not 100% sure what solved it, but it does not come up for me anymore. I moved towards publishing all packages of my project to a private registry, because other tools require this also. However, I also remember this issue coming up again at some point and I chose to ignore it, because my focus was the app. I believe this was before I switched to a private registry.

**Comment by itsJASPERr** [comment]
It just came to my mind: it is the Babel transpiration. If you include the packages in Babel transpiration, it will work for web. I think there are multiple ways, either use Babel plugin directly, or add the list of packages you need expo to transpire to the app.json. On my phone right now, I can look up the confit later (or see app config docs)

**Comment by andrioid** [comment]
That makes sense. We have a CRA setup currently with React Rewired and we manually include our own packages into the webpack babel section.

Maybe we need something similar here. I’d love for this not to be a hack.

Any ideas @EvanBacon?

**Comment by EvanBacon** [comment]
Resolved in #12566
